app/UserService.ts

```diff
@@ -89,10 +98,22 @@ export class UserService extends ChordNode {
     server.start();
   }
 
-  // Removes a User from local state
-  removeUser(id) {
-    if (this.userMap[id]) {
-      delete this.userMap[id];
+  // Streams a List of User IDs stored by the Node
+  getUserIds(call) {
+    const users = Object.values(this.userMap);
+    users.forEach(user => {
+      call.write({
+        id: user.id,
+        metadata: user.metadata
+      });
+    });
+    call.end();
+  }
+
+  // Removes a User from local state matching the hash
+  removeUser(hashedUserId) {
+    if (this.userMap[hashedUserId]) {
```

Comment:
Hmm... I don't know if this is right.  I think we want to delete by the unique ID rather than the hashed ID since the latter is guaranteed to have collisions and is really only used to figure out "where" (i.e., what node) to store the non-hashed ID - index card versus drawer.  What I'm thinking is that if there is a collision, which one is going to get removed: the one you wanted to remove, all of the ones with the same hashed ID, neither of them, the first one in the list of matches?